library/ziti_ctrl.c

```diff
@@ -844,4 +844,52 @@ void ziti_ctrl_post_mfa_recovery_codes(ziti_controller *ctrl, char *body, size_t
     um_http_req_t *req = start_request(ctrl->client, "POST", "/current-identity/mfa/recovery-codes", ctrl_resp_cb, resp);
     um_http_req_header(req, "Content-Type", "application/json");
     um_http_req_data(req, body, body_len, free_body_cb);
+}
+
+void ziti_ctrl_extend_cert_authenticator(ziti_controller *ctrl, char *authenticatorId, char *csr, void(*cb)(ziti_extend_cert_authenticator_resp*, const ziti_error *, void *), void *ctx) {
+    if(!verify_api_session(ctrl, cb, ctx)) return;
```

Comment:
if there is an error, callback should be called. Caller may need to free resources